### PR TITLE
BUGFIX: `XmlResponseFormatter` element names with namespace

### PR DESCRIPTION
**Pull request by WinterSilence**
- Fix for extended(with namespace) element names:
~~~php
$dom = new DOMDocument();
$el = $dom->createElement('image:image', 'url'); // Ok
$el = new DOMElement('image:image', 'url'); // DOMException: Namespace Error
~~~
`image:*` tags uses in [Google image sitemap](https://developers.google.com/search/docs/advanced/sitemaps/image-sitemaps).
- Added property `$objectTagToLowercase` to convert tag name to lowercase on `$useObjectTags`
- Small updates in comments

| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  |

### CODE REVIEW

File: tests/framework/web/XmlResponseFormatterTest.php [hunk 1]
```diff
@@ -57,7 +58,8 @@ public function formatArrayDataProvider()
             [[
                 'a' => 1,
                 'b' => 'abc',
-            ], "<response><a>1</a><b>abc</b></response>\n"],
+                'image:loc' => 'url',
+            ], "<response><a>1</a><b>abc</b><image:loc>url</image:loc></response>\n"],
```
Comment (samdark) [inline tests/framework/web/XmlResponseFormatterTest.php:62]: That changes format significantly. Likely will result in broken apps.

File: tests/framework/web/XmlResponseFormatterTest.php [hunk 2]
```diff
@@ -79,7 +81,7 @@ public function formatArrayDataProvider()
                 'b:c' => 'b:c',
                 'a b c' => 'a b c',
                 'äøñ' => 'äøñ',
-            ], "<response><item>1</item><item>2015-06-18</item><item>b:c</item><item>a b c</item><äøñ>äøñ</äøñ></response>\n"],
+            ], "<response><item>1</item><item>2015-06-18</item><b:c>b:c</b:c><item>a b c</item><äøñ>äøñ</äøñ></response>\n"],
```
Comment (samdark) [inline tests/framework/web/XmlResponseFormatterTest.php:84]: That changes format significantly. Likely will result in broken apps.

File: tests/framework/web/XmlResponseFormatterTest.php (hunk 1 above)
Comment (WinterSilence) [reply tests/framework/web/XmlResponseFormatterTest.php:62]: @samdark uses `item` tag instead `image:loc` that's I'm calling "broken app".

File: tests/framework/web/XmlResponseFormatterTest.php (hunk 2 above)
Comment (WinterSilence) [reply tests/framework/web/XmlResponseFormatterTest.php:84]: @samdark uses `item` tag instead `b:c` that's I'm calling "broken app".

Comment (samdark) [reply tests/framework/web/XmlResponseFormatterTest.php:84]: This part looks alright.

**Comment by samdark** [comment]
Bugfix part of this PR is likely correct.

File: framework/web/XmlResponseFormatter.php [hunk 3]
```diff
@@ -26,87 +25,112 @@
...
+     * @var string the XML version.
      */
     public $version = '1.0';
     /**
      * @var string the XML encoding. If not set, it will use the value of [[Response::charset]].
      */
     public $encoding;
     /**
-     * @var string the name of the root element. If set to false, null or is empty then no root tag should be added.
+     * @var string|string[]|false the name of the root element or index array (URI namespace, tag name).
```
Comment (samdark) [inline framework/web/XmlResponseFormatter.php:40]: What is index array?

File: framework/web/XmlResponseFormatter.php [hunk 4]
```diff
@@ -26,87 +25,112 @@
...
+     * If set to false or null then no root tag should be added.
      */
     public $rootTag = 'response';
     /**
-     * @var string the name of the elements that represent the array elements with numeric keys.
+     * @var string the name of the elements that represent the array elements with numeric keys
```
Comment (samdark) [inline framework/web/XmlResponseFormatter.php:45]:
```suggestion
     * @var string the name of the elements that represent the array elements with numeric keys.
```

File: framework/web/XmlResponseFormatter.php [hunk 5]
```diff
@@ -26,87 +25,112 @@
...
      */
     public $itemTag = 'item';
     /**
-     * @var bool whether to interpret objects implementing the [[\Traversable]] interface as arrays.
-     * Defaults to `true`.
+     * @var bool whether to interpret objects implementing the [[Traversable]] interface as arrays
```
Comment (samdark) [inline framework/web/XmlResponseFormatter.php:49]:
```suggestion
     * @var bool whether to interpret objects implementing the [[\Traversable]] interface as arrays.
     * Defaults to `true`.
```

File: framework/web/XmlResponseFormatter.php (hunk 3 above)
Comment (WinterSilence) [reply framework/web/XmlResponseFormatter.php:40]: @samdark
> or index array (URI namespace, tag name).

File: framework/web/XmlResponseFormatter.php [hunk 6]
```diff
@@ -52,61 +52,86 @@ class XmlResponseFormatter extends Component implements ResponseFormatterInterfa
...
     /**
      * Formats the specified response.
+     *
      * @param Response $response the response to be formatted.
      */
     public function format($response)
     {
-        $charset = $this->encoding === null ? $response->charset : $this->encoding;
+        if ($this->encoding === null) {
+            $this->encoding = $response->charset;
```
Comment (samdark) [inline framework/web/XmlResponseFormatter.php:80]: Causes side effect in the method. Let's revert this part.

Comment (WinterSilence) [reply framework/web/XmlResponseFormatter.php]: @samdark reverted

File: framework/web/XmlResponseFormatter.php [hunk 7]
```diff
@@ -95,18 +115,22 @@ protected function buildXml($element, $data)
...
                     $element->appendChild($child);
-                    $child->appendChild(new DOMText($this->formatScalarValue($value)));
                 }
             }
         } elseif (is_object($data)) {
             if ($this->useObjectTags) {
-                $child = new DOMElement(StringHelper::basename(get_class($data)));
+                $name = StringHelper::basename(get_class($data));
+                if ($this->objectTagToLowercase) {
+                    $name = strtolower($name);
```
Comment (samdark) [inline framework/web/XmlResponseFormatter.php:131]: Won't work well with UTF.

Comment (WinterSilence) [reply framework/web/XmlResponseFormatter.php:131]: @samdark as Yii at all, for example https://github.com/yiisoft/yii2/blob/bcd5b48492219efc58e1d11c9ddb61c0df0dc370/framework/console/controllers/MigrateController.php#L409

**Review by samdark — APPROVED** [review]

**Review by bizley — APPROVED** [review]

**Comment by WinterSilence** [comment]
@samdark I'm will try be shortest

**Pull request merged**